### PR TITLE
Set $CARGO_HOME, $PATH and move cfg under /tmp

### PR DESCRIPTION
**Pull request by idelvall**

### CODE REVIEW

File: rust/Earthfile [hunk 1]
```diff
@@ -13,6 +13,9 @@ INIT:
     IF [ "$CARGO_HOME" = "" ]
         ENV CARGO_HOME="$HOME/.cargo"
     END
+    IF ! echo "$PATH" | grep -q "$CARGO_HOME/bin"
```
Comment (xv-ian-c) [inline rust/Earthfile:16]: In my `.bashrc` I have this construct as:
```
if ! echo $PATH | /bin/egrep -q "(^|:)$1($|:)" ; then
```

Which avoids matching a path which happens to contain but is not exactly `$CARGO_HOME/bin`. Maybe doesn't matter for this specific use but though I'd would mention it.

Comment (xv-ian-c) [reply rust/Earthfile:16]: FWIW I think `grep -E` is the same as `egrep` (which indeed might be a Debian specific synonym)?

Comment (idelvall) [reply rust/Earthfile:16]: yes, but my idea was that that synonym is not generally available though.

Comment (xv-ian-c) [reply rust/Earthfile:16]: Sure, was just saying you can probably just use `/bin/grep -Eq "(^|:)$1($|:)" `

Comment (idelvall) [reply rust/Earthfile]: I see how I expressed myself badly :)
ANyway it's pushed!
Thanks!

File: rust/Earthfile [hunk 2]
```diff
@@ -6,27 +6,33 @@ VERSION --global-cache 0.7
 #   - sweep_days (4): +CARGO calls use cargo-sweep to clean build artifacts that haven't been accessed for this number of days.
 INIT:
     COMMAND
-    RUN if [ -f /earthly/cfg/cache_id ]; then \
+    RUN if [ -f /tmp/earthly/cfg/cache_id ]; then \
```
Comment (xv-ian-c) [inline rust/Earthfile:9]: Should `/earthly_lib_rust_temp` move too?
Also I still see a `ENV CARGO_HOME="/earthly/.cargo"`

Comment (idelvall) [reply rust/Earthfile:9]: Fixed

**Review by xv-ian-c — APPROVED** [review]

**Pull request merged**